lib/veri_traci.ml

```diff
@@ -0,0 +1,89 @@
+open Bap.Std
+module Bap_value = Value
+open Bap_traces.Std
+open Monad.State
+open Bil.Result
+
+type event = Trace.event
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+(** TODO: what is a behaviour of update_event method?  *)
+class context arch = object(self:'s)
+  inherit Bili.context
+  val mem  : var = mem_of_arch arch
+  val arch : arch = arch
+  method arch = arch
+  method mem = mem
+  method update_event: event -> 's = fun ev -> failwith "unimplemented"
+end
+
+let mem_operands mv = 
+  Size.of_int_opt (Bitvector.bitwidth (Move.data mv)) |> function
+  | None -> None
+  | Some sz ->
+    let addr = Bil.int (Move.cell mv) in
+    let data = Bil.int (Move.data mv) in
+    Some (addr, data, sz)
+
+class ['a] t = object(self)
+  constraint 'a = #context
+  inherit ['a] Bili.t
+
+  method private eval_reg: var move -> 'a u = fun mv ->  
+    self#eval_move (Move.cell mv) (Bil.int (Move.data mv))
+
+  method private eval_mem_op op mv : 'a r =
+    match mem_operands mv with 
+    | None -> 
+      get () >>= fun ctxt -> 
+      let (c,r) = ctxt#create_undefined in
+      put c >>= fun () -> return r
+    | Some (addr, data, size) ->
+      get () >>= fun ctxt ->
+      let endian = Arch.endian ctxt#arch in
+      match op with
+      | `Store ->
+        self#eval_store ~mem:(Bil.var ctxt#mem) ~addr data endian size
+      | `Load ->
+        self#eval_load ~mem:(Bil.var ctxt#mem) ~addr endian size
+
+  method eval_memory_store mv = 
+    self#eval_mem_op `Store mv >>= fun r ->
+    get () >>= fun ctxt -> self#update ctxt#mem r
+
+  method eval_memory_load: addr move -> 'a u = fun mv -> 
```

Comment:
The idea was, is that we have a memory load operation, then we should first put the expected value into the memory, and afterwards load it. Like this: https://www.youtube.com/watch?v=Q-e2urqArko